include/sys/dnode.h

```diff
@@ -202,6 +221,7 @@ typedef struct dnode {
 	uint32_t dn_datablksz;		/* in bytes */
 	uint64_t dn_maxblkid;
 	uint8_t dn_next_type[TXG_SIZE];
+	uint8_t dn_count;		/* metadnode slots consumed on disk */
```

Comment:
good that there's a comment explaining what this is, but even better would be to name the variable more evocatively.  e.g. dn_num_slots